### PR TITLE
Reorganized less variables for simpler theming

### PR DESCRIPTION
**Pull request by mnapoli**
See #7876 for background

We now have `theme.less` and `theme-advanced.less` to separate simple variables for plugin developers from advanced variables.

In Morpheus:

```
base.less (unchanged)
main.less (was previously theme.less)
theme.less -> variables for themes (was previously `less/_variables.less`)
theme-advanced.less -> advanced variables for themes
```

That makes Morpheus have the same layout as other themes (i.e. `theme.less` contains the theme variables), which I think is great for developers that could be previously confused with the file names.

By the way I took great care to make sure git moves files and not deletes/creates them, but the diff here doesn't seem to show that which is a shame (in phpstorm files are correctly moved) -_-

### CODE REVIEW

File: plugins/Morpheus/stylesheets/theme-advanced.less [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@theme-color-background-base:          #fff;
+@theme-color-background-tinyContrast:  #f2f2f2;
+@theme-color-background-lowContrast:   @color-silver-l80;
+@theme-color-background-contrast:      #5F5A60;
+@theme-color-background-highContrast:  #202020;
+
+@theme-color-border: @color-silver-l80;
+
+@theme-color-menu-contrast-text: @theme-color-text-lighter;
```
Comment (tsteur) [inline plugins/Morpheus/stylesheets/theme-advanced.less]: `@theme-color-menu-*` and `@theme-color-widget-*` should be in `theme.less` I think. Those are specifically exposed API's/variables to allow developers to change those. Or with other words those are the things we identified that companies might want to change if they want to adjust Piwik to their CI.

**Comment by tsteur** [comment]
For me there are 2 things still unclear:

1. How do we make clear that only variables in `theme.less` and `theme-advanced.less` are API and BC is guaranteed. We might need a new page under developer.piwik.org -> `Reference`? Maybe this one could be automatically created with the other pages? It could automatically detect a doc block for each variable and generate documentation based on this.
2. How do we make sure we keep BC etc? Can we maybe create a UI test that activates a theme and takes only a very few screenshots eg of dashboard, a report page, an admin page to see whether we break anything and to make sure when adding new colors/css that it will work with this? (It is quite easy to break theming eg just by renaming the files eg because of https://github.com/piwik/piwik/blob/2.14.0-b1/core/AssetManager/UIAssetFetcher/StylesheetUIAssetFetcher.php#L16-L42 , we had this in the past)

Asking 1) here as we added a couple of new variables and I would not feel good about adding more variables without having this. We can create a new issue for automatically generating it, but then we should maybe create at least a developer documentation/reference page manually.

**Comment by mnapoli** [comment]
1) We could update the developer documentation (i.e. link to those files), that's what I was thinking at least.
2) Sounds like a good idea (to be done separately).

FYI I haven't added or removed variables in this PR, I have only reorganized them. And I believe since I started the redesign I have only added the `@theme-color-border` variable.

File: plugins/Morpheus/stylesheets/theme-advanced.less (hunk 1 above)
Comment (mnapoli) [inline plugins/Morpheus/stylesheets/theme-advanced.less]: OK, last time though you argued that most themes are about replacing the main Piwik colors with the one of the company… Anyway I don't mind moving those variables.

Comment (mnapoli) [inline plugins/Morpheus/stylesheets/theme-advanced.less]: Done

Comment (tsteur) [inline plugins/Morpheus/stylesheets/theme-advanced.less]: Not sure where I mentioned it, eg in https://github.com/piwik/piwik/issues/7876#issuecomment-100704490 I mentioned to put the menu and widget colors in `theme.less`. They are very important for theming. The `@theme-color-background-*` variables should be still in `theme-advanced` as they are "complicated" (whch I also mentioned here https://github.com/piwik/piwik/pull/7793#issuecomment-98549220 ). 

I know they default to other variables but we want users to be able to just specifically change those widget and menu colors without affecting anything else to keep it as easy as possible. Just by changing those 5 menu and widget variables they will be able to create a nice theme according to their CI.

**Comment by mnapoli** [comment]
The UI test fail is a random failure: http://builds-artifacts.piwik.org/ui-tests.simpler-theme/12676.7/screenshot-diffs/diffviewer.html

Ready to be merged

**Comment by mattab** [comment]
Feedback:
 * Simpler theming! :+1: 
 * we need to update the Theming developer guide which references theme.less file http://developer.piwik.org/guides/theming
 * Do you think this potentially could break any of the two themes published on the Marketplace  (and possibly other themes?) or is BC kept?

**Comment by mnapoli** [comment]
I have just renamed some files and moved less variables, so there should be no BC breaks.

Will update the developer documentation.

**Comment by tsteur** [comment]
> I have just renamed some files and moved less variables, so there should be no BC breaks.

We should still test it if possible. As mentioned earlier we had problems in the past with this just by renaming files eg because of 
https://github.com/piwik/piwik/blob/2.14.0-b1/core/AssetManager/UIAssetFetcher/StylesheetUIAssetFetcher.php#L16-L42

**Pull request merged**